synapse/app/homeserver.py

```diff
@@ -234,7 +234,7 @@ def _configure_named_resource(self, name, compress=False):
             )
 
         if name in ["media", "federation", "client"]:
-            if self.config.media.enable_media_repo:
+            if self.config.server.enable_media_repo:
```

Comment:
I might be missing context here, but why was this line (wrongly) trying to get the setting from the `media` config? Ditto for the `account_validity` vs `email` config.

Also, and not something that needs to be fixed now, I find it a bit worrying if we have been looking for this setting in the wrong config class for some time and CI didn't pick it up.

Reply:
> I might be missing context here, but why was this line (wrongly) trying to get the setting from the `media` config? Ditto for the `account_validity` vs `email` config.

In one of the previous PRs I updated it to look at the wrong configuration.

> Also, and not something that needs to be fixed now, I find it a bit worrying if we have been looking for this setting in the wrong config class for some time and CI didn't pick it up.

It still worked fine before this PR, which is why CI didn't pick it up. It is due to this code:

https://github.com/matrix-org/synapse/blob/94b620a5edd6b5bc55c8aad6e00a11cc6bf210fa/synapse/config/_base.py#L121-L134

(Which I need to double check is removed by this PR actually...)

Reply:
Yep, I removed that! 🎉